horace_core/sqw/@sqw/get_write_handle.m

```diff
@@ -7,6 +7,9 @@
 % Optional input:
 % outfile  -- target sqw file will be created with the filename provided
 %             If missing, the file will have random temporary filename
+% outfile_defined
+%          -- if provided and true, do not generate tmp filename for
+%             operation but use the one provided.
```

Comment:
The logic here is a bit more tricky. filename is always provided, but operation (which is usually long) is performed to tmp file to be deleted if the operation is canceled. The original file is left untached to avoid issues with IO. After that, depending on the algorithm settings, either new object is created backed by tmp file constructed from initial file, or initial file is deleted and the object is moved to sit over the new tmp file or new non-tmp file is the result. In two last cases this switch is used.